feature-libs/customer-ticketing/components/details/customer-ticketing-messages/customer-ticketing-messages.component.ts

```diff
@@ -0,0 +1,125 @@
+/*
+ * SPDX-FileCopyrightText: 2022 SAP Spartacus team <spartacus-team@sap.com>
+ * SPDX-FileCopyrightText: 2023 SAP Spartacus team <spartacus-team@sap.com>
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+import { Component, OnDestroy, ViewChild } from '@angular/core';
+import { EventService } from '@spartacus/core';
+import {
+  MessageEvent,
+  MessagingComponent,
+  MessagingConfigs,
+} from '@spartacus/storefront';
+import {
+  CustomerTicketingConfig,
+  CustomerTicketingFacade,
+  STATUS,
+  TicketDetails,
+  TicketEvent,
+  GetTicketQueryReloadEvent,
+} from '@spartacus/customer-ticketing/root';
+import { Observable, Subscription } from 'rxjs';
+import { map } from 'rxjs/operators';
+@Component({
+  selector: 'cx-customer-ticketing-messages',
+  templateUrl: './customer-ticketing-messages.component.html',
+})
+export class CustomerTicketingMessagesComponent implements OnDestroy {
+  @ViewChild(MessagingComponent) messagingComponent: MessagingComponent;
+
+  ticketDetails$: Observable<TicketDetails | undefined> =
+    this.customerTicketingFacade.getTicket();
+
+  constructor(
+    protected customerTicketingConfig: CustomerTicketingConfig,
+    protected customerTicketingFacade: CustomerTicketingFacade,
+    protected eventService: EventService
+  ) {}
+
+  subscription = new Subscription();
+
+  messageEvents$: Observable<Array<MessageEvent>> = this.prepareMessageEvents();
+
+  messagingConfigs: MessagingConfigs = this.prepareMessagingConfigs();
+
+  onSend(event: { files: FileList | File | undefined; message: string }) {
+    this.subscription.add(
+      this.customerTicketingFacade
+        .createTicketEvent(this.prepareTicketEvent(event.message))
+        .subscribe((createdEvent: TicketEvent) => {
+          if (
+            event.files instanceof FileList &&
+            event.files?.length &&
+            createdEvent.code
+          ) {
+            this.customerTicketingFacade.uploadAttachment(
+              event.files.item(0) as File,
+              createdEvent.code
+            );
+          } else {
+            this.eventService.dispatch({}, GetTicketQueryReloadEvent);
```

Comment:
I am dispatching this event here because, we don't want ticket data to reload if any attachment is attached. as we would end up ticket data twice with the attachment.